sdks/python/apache_beam/coders/coders.py

```diff
@@ -741,6 +742,36 @@ def __hash__(self):
     return hash(type(self))
 
 
+class _MemoizingPickleCoder(_PickleCoderBase):
+  """Coder using Python's pickle functionality with memoization."""
+  def __init__(self, cache_size=16):
+    super(_MemoizingPickleCoder, self).__init__()
+    self.cache_size = cache_size
+
+  def _create_impl(self):
+    dumps = pickle.dumps
```

Comment:
Should we be using pickler rather than raw pickle here?

Reply:
makes sense. I've put pickler in, since the current use (`BoundedSource` elements) contains code and context.